src/Microsoft.ML.ImageAnalytics/ImageResizer.cs

```diff
@@ -292,13 +292,18 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                 ValueGetter<Bitmap> del =
                     (ref Bitmap dst) =>
                     {
-                        if (dst != null)
-                            dst.Dispose();
-
                         getSrc(ref src);
                         if (src == null || src.Height <= 0 || src.Width <= 0)
                             return;
 
+                        if (src.Height == info.ImageHeight && src.Width == info.ImageWidth)
+                        {
+                            dst = src;
+                            return;
+                        }
+                        else if(dst != null)
```

Comment:
When `getSrc` gets called more than once, the src may already be populated (`ref dst = ref src` ). This is passed to ImageResizer and the reference gets disposed before trying to retrieve it again. A way to resolve issue is to only dispose `src` only if it will be overwritten by a new Bitmap.